### ISSUE TITLE
RuntimeWarning: divide by zero

### ISSUE THREAD

**Issue by ldv1** [numericalalgorithmsgroup/pybobyqa#9]
Hi,
first of all, big thanks for Py-BOBYQA!

I started with something simple I found [here](https://people.maths.ox.ac.uk/robertsl/pybobyqa/userguide.html):

```
import numpy as np
import pybobyqa

def rosenbrock(x):
    return 100.0 * (x[1] - x[0] ** 2) ** 2 + (1.0 - x[0]) ** 2

def rosenbrock_noisy(x):
    return rosenbrock(x) * (1.0 + 1e-2 * np.random.normal(size=(1,))[0])

x0 = np.array([-1.2, 1.0])
np.random.seed(0)
soln = pybobyqa.solve(rosenbrock_noisy, x0, objfun_has_noise=True)
print(soln)
```
Solution is correct, but I got

```
/usr/lib/python3.6/site-packages/pybobyqa/solver.py:510: RuntimeWarning: divide by zero encountered in log
  np.log(restart_auto_detect_chg_grad))
/usr/lib64/python3.6/site-packages/numpy/lib/function_base.py:3103: RuntimeWarning: invalid value encountered in subtract
  X -= avg[:, None]
/usr/lib/python3.6/site-packages/pybobyqa/solver.py:513: RuntimeWarning: divide by zero encountered in log
  np.log(restart_auto_detect_chg_hess))
```

Is it something I should worry about?
thanks for the help,
regards,
Laurent

**Comment by lindonroberts** [comment]
Thanks Laurent - this happens when Py-BOBYQA is trying to  determine when to call its restart functionality. The worst-case scenario should be that Py-BOBYQA works, but uses a few more objective evaluations. I will fix this issue

**Comment by lindonroberts** [comment]
Fixed in v1.2
